src/generator/qbe.rs

```diff
@@ -0,0 +1,596 @@
+    Sgt,
+    /// Returns 1 if first value is greater than or equal to second, respecting signedness
+    Sge,
+    /// Returns 1 if values are equal
+    Eq,
+    /// Returns 1 if values are not equal
+    Ne,
+}
+
+/// QBE instruction
+#[derive(Debug)]
+enum QbeInstr {
+    /// Adds values of two temporaries together
+    Add(QbeTemporary, QbeTemporary),
+    /// Subtracts the second value from the first one
+    Sub(QbeTemporary, QbeTemporary),
+    /// Multiplies values of two temporaries
+    Mul(QbeTemporary, QbeTemporary),
+    /// Divides the first value by the second one
+    Div(QbeTemporary, QbeTemporary),
+    /// Returns a remainder from division
+    Rem(QbeTemporary, QbeTemporary),
+    /// Performs a comparion between values
+    Cmp(QbeType, QbeCmp, QbeTemporary, QbeTemporary),
+    /// Performs a bitwise AND on values
+    And(QbeTemporary, QbeTemporary),
+    /// Performs a bitwise OR on values
+    Or(QbeTemporary, QbeTemporary),
+    /// Copies either a temporary or a literal value
+    Copy(Either<QbeTemporary, usize>),
+    /// Return from a function, optionally with a value
+    Ret(Option<QbeTemporary>),
+}
+
+impl fmt::Display for QbeInstr {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Self::Add(lhs, rhs) => write!(f, "add {}, {}", lhs, rhs),
+            Self::Sub(lhs, rhs) => write!(f, "sub {}, {}", lhs, rhs),
+            Self::Mul(lhs, rhs) => write!(f, "mul {}, {}", lhs, rhs),
+            Self::Div(lhs, rhs) => write!(f, "div {}, {}", lhs, rhs),
+            Self::Rem(lhs, rhs) => write!(f, "rem {}, {}", lhs, rhs),
+            Self::Cmp(ty, cmp, lhs, rhs) => {
+                assert!(
+                    !matches!(ty, QbeType::Aggregate(_)),
+                    "Cannot compare aggregate types"
+                );
+
+                write!(
+                    f,
+                    "c{}{} {}, {}",
+                    match cmp {
+                        QbeCmp::Slt => "slt",
+                        QbeCmp::Sle => "sle",
+                        QbeCmp::Sgt => "sgt",
+                        QbeCmp::Sge => "sge",
+                        QbeCmp::Eq => "eq",
+                        QbeCmp::Ne => "ne",
+                    },
+                    ty,
+                    lhs,
+                    rhs,
+                )
+            }
+            Self::And(lhs, rhs) => write!(f, "and {}, {}", lhs, rhs),
+            Self::Or(lhs, rhs) => write!(f, "or {}, {}", lhs, rhs),
+            Self::Copy(val) => match val {
+                Either::Left(temp) => write!(f, "copy {}", temp),
+                Either::Right(lit) => write!(f, "copy {}", *lit),
+            },
+            Self::Ret(val) => match val {
+                Some(val) => write!(f, "ret {}", val),
+                None => write!(f, "ret"),
+            },
+        }
+    }
+}
+
+/// QBE type
+#[derive(Debug, Eq, PartialEq, Clone)]
+#[allow(dead_code)]
```

Comment:
Why is this lint ignored?

Reply:
Because not all types are in use right now, and having a ton of warnings on every build is annoying